src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/Stream.DeserializeAsyncEnumerable.cs

```diff
@@ -73,12 +74,74 @@ public static async Task DeserializeAsyncEnumerable_ReadSourceAsync<TElement>(IE
             Assert.Equal(source, results);
         }
 
+        [Fact]
+        public static async Task DeserializeAsyncEnumerable_ShouldTolerateCustomQueueConverters()
+        {
+            const int expectedCount = 20;
+
+            JsonSerializerOptions options = new JsonSerializerOptions
+            {
+                Converters = { new DegenerateQueueConverterFactory() }
+            };
+
+            byte[] data = JsonSerializer.SerializeToUtf8Bytes(Enumerable.Repeat(Enumerable.Repeat(1,3), expectedCount));
+
+            using var stream = new MemoryStream(data);
+
+            int callbackCount = 0;
+            await foreach (Queue<int> nestedQueue in JsonSerializer.DeserializeAsyncEnumerable<Queue<int>>(stream, options))
+            {
+                Assert.Equal(1, nestedQueue.Count);
+                Assert.Equal(0, nestedQueue.Peek());
+                callbackCount++;
+            }
+
+            Assert.Equal(expectedCount, callbackCount);
+        }
+
+        private class DegenerateQueueConverterFactory : JsonConverterFactory
```

Comment:
Why take a factory + generics approach if we aren't testing multiple `<T>`'s?

Reply:
If you take a look at the test, the deserialized value is of type `IAsyncEnumerable<Queue<int>>`, so the value is actually being deserialized as `Queue<Queue<int>>`. This is to test that the outer queue uses the hardcoded converter but the inner queues still reverts to the user supplied custom converter. For this to be checked the custom converter needs to support both `Queue<int>` and `Queue<Queue<int>>` types.